go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -283,8 +299,12 @@ func getPodDetails(pod *kapi.Pod, nodeName string) ([]*net.IPNet, net.HardwareAd
 func podChanged(pod1 *kapi.Pod, pod2 *kapi.Pod, nodeName string) bool {
 	podIPs1, mac1, _ := getPodDetails(pod1, nodeName)
 	podIPs2, mac2, _ := getPodDetails(pod2, nodeName)
+	podExGw1 := pod1.GetAnnotations()[hotypes.HybridOverlayExternalGw]
```

Comment:
I think @dcbw and @danwinship were suggesting to stop using GetAnnotations and just directly access the annotations

Reply:
yes. `GetAnnotations` is only needed when you're operating on a `runtime.Object` of unknown type

Reply:
fixed